libs/vm-modules/include/vm_modules/math/tensor/tensor_estimator.hpp

```diff
@@ -275,6 +275,9 @@ class TensorEstimator
   static ChargeAmount ToChargeAmount(fixed_point::fp64_t const &val);
 
   VMObjectType &tensor_;
+
+  DataType padded_size_;
```

Comment:
I think you should ditch this.

I understand that this reduces the number of lines of code in the estimator implementation but it also means that the estimator is doing some of the actual work of the tensor, which is fundamentally not what estimators should be doing (i.e. you would need to update these values when you reshape/resize the tensor, and now you have to ensure estimator always tracks its tensor size/shape properly). You would also need to serialise and deserialise these two values (which I think you missed?).

Reply:
fixed